pinot-common/src/main/java/org/apache/pinot/common/restlet/resources/SegmentStatus.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.restlet.resources;
+
+import java.util.Objects;
+
+public class SegmentStatus {
```

Comment:
SegmentStatus is used in the server code as well. So, added a class to common for referencing.

Reply:
Working on changing the equals and hashcode. Will commit the changes.

Reply:
Actually, we may not need the equality methods yet, removing them.

Reply:
I strongly suggest we return a different object to the user instead of returning this one. It will enable us to evolve the internal interface independent of interface exposed to the user.
Also, the controller returned status could include other run-time status about a segment. For example:
1. Helix externalview of the server so we know if it is online or not.
2. The crc, size, date uploaded/refreshed, etc. from segment metadata
3. Other info as wee want to add later (e.g. number of times segment was hit/searched/selected since reboot, etc.)

Reply:
Please document each member in this object clearly, what it contains in various situations

Reply:
Thanks for the suggestions! I will think about how to refactor my code and commit again.